Cabal/src/Distribution/Simple/GHC/Internal.hs

```diff
@@ -505,12 +505,13 @@ componentJsGhcOptions verbosity lbi bi clbi odir filename =
       ghcOptVerbosity = toFlag (min verbosity normal)
     , ghcOptMode = toFlag GhcModeCompile
     , ghcOptInputFiles = toNubListR [filename]
+    , ghcOptJSppOptions = jsppOptions bi
     , ghcOptCppIncludePath = includePaths lbi bi clbi odir
     , ghcOptHideAllPackages = toFlag True
     , ghcOptPackageDBs = withPackageDB lbi
     , ghcOptPackages = toNubListR $ mkGhcOptPackages (promisedPkgs lbi) clbi
     , ghcOptObjDir = toFlag odir
-    , ghcOptExtra = hcOptions GHC bi
+    , ghcOptExtra = jsppOptions bi <> hcOptions GHC bi
```

Comment:
This looks surprising? I would not expect to pass these options using `-optJSP` and normally on the command line.

Reply:
it's true, i'll remove it